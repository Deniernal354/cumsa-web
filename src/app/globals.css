@import "tailwindcss";

:root {
  --background: #f8fafc;
  --foreground: #0f172a;
  --surface: #ffffff;
  --muted: #64748b;
  --navy: #2a7487;
  --blue: #5aced8;
  --red: #c73221;
  --header-top: #123a66;
}

body {
  background: var(--background);
  color: var(--foreground);
  font-family: Arial, Helvetica, sans-serif;
}

a {
  text-underline-offset: 4px;
}

.link-cta {
  color: var(--blue);
  font-weight: 700;
  text-decoration: underline;
  text-underline-offset: 3px;
}

.link-cta:hover {
  color: var(--red);
}

.btn-cozy {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 0.5rem 1rem;
  border-radius: 0.6rem;
  font-weight: 600;
  font-size: 0.9rem;
  transition:
    transform 150ms ease,
    box-shadow 150ms ease,
    opacity 150ms ease;
}

.btn-cozy:hover {
  transform: translateY(-1px);
  box-shadow: 0 6px 16px rgba(0, 0, 0, 0.08);
}

.btn-primary {
  background: var(--blue);
  color: white;
}

.btn-outline {
  background: var(--surface);
  color: var(--foreground);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

:focus-visible {
  outline: 2px solid color-mix(in oklab, var(--blue) 70%, white);
  outline-offset: 2px;
}

.container-base {
  margin-left: auto;
  margin-right: auto;
  max-width: 80rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

@media (min-width: 640px) {
  .container-base {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media (min-width: 1024px) {
  .container-base {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}

.transition-base {
  transition: all 200ms ease-out;
}

.card {
  background: var(--surface);
  border: 1px solid rgba(0, 0, 0, 0.12);
  border-radius: 0.75rem;
}

.title {
  color: var(--red);
}

.tint-blue {
  filter: invert(28%) sepia(88%) saturate(3612%) hue-rotate(211deg) brightness(93%) contrast(98%);
}

.logo-mask-blue {
  background-color: var(--blue);
  -webkit-mask-image: url("/CU-MSA-LOGO-blank-2.png");
  mask-image: url("/CU-MSA-LOGO-blank-2.png");
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
  -webkit-mask-position: center;
  mask-position: center;
  -webkit-mask-size: contain;
  mask-size: contain;
  display: inline-block;
}
