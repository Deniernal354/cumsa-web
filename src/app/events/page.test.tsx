import { test, expect } from "@jest/globals";
import { render, act } from "@testing-library/react";
import EventsPage from "./page";

// Mock fetch to avoid network calls in tests
const mockFetch = jest.fn(() =>
  Promise.resolve({
    ok: true,
    json: () => Promise.resolve([]), // Return empty array for events
  } as Response),
);

global.fetch = mockFetch;

test("events page snapshot", async () => {
  let result: ReturnType<typeof render>;

  await act(async () => {
    result = render(<EventsPage />);
    // Wait for useEffect to complete
    await new Promise((resolve) => setTimeout(resolve, 0));
  });

  expect(result!.asFragment()).toMatchSnapshot();
});
