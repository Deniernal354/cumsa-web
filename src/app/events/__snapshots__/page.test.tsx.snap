// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`events page snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-base py-12"
  >
    <div
      class="portal-glow"
    >
      <div
        class="w-full p-8 space-y-8"
      >
        <div
          class="text-center space-y-4"
        >
          <div
            class="inline-flex items-center gap-2 px-4 py-2 bg-blue-50 rounded-full text-blue-700 font-medium text-sm"
          >
            <svg
              class="w-4 h-4"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="384"
                rx="48"
                stroke-linejoin="round"
                stroke-width="32"
                width="416"
                x="48"
                y="80"
              />
              <circle
                cx="296"
                cy="232"
                r="24"
              />
              <circle
                cx="376"
                cy="232"
                r="24"
              />
              <circle
                cx="296"
                cy="312"
                r="24"
              />
              <circle
                cx="376"
                cy="312"
                r="24"
              />
              <circle
                cx="136"
                cy="312"
                r="24"
              />
              <circle
                cx="216"
                cy="312"
                r="24"
              />
              <circle
                cx="136"
                cy="392"
                r="24"
              />
              <circle
                cx="216"
                cy="392"
                r="24"
              />
              <circle
                cx="296"
                cy="392"
                r="24"
              />
              <path
                d="M128 48v32m256-32v32"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="32"
              />
              <path
                d="M464 160H48"
                fill="none"
                stroke-linejoin="round"
                stroke-width="32"
              />
            </svg>
            Upcoming Events
          </div>
          <h1
            class="text-4xl font-bold text-gray-900 tracking-tight"
          >
            Join Our Amazing 
            <span
              class="text-transparent bg-clip-text bg-gradient-to-r from-blue-600 to-purple-600"
            >
              Events
            </span>
          </h1>
          <p
            class="text-lg text-gray-600 max-w-2xl mx-auto"
          >
            Discover exciting opportunities to connect, learn, and grow with our community!
          </p>
        </div>
        <div
          class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-6"
        >
          <div
            class="col-span-full flex flex-col items-center justify-center py-16 text-center"
          >
            <div
              class="w-16 h-16 bg-gray-100 rounded-full flex items-center justify-center mb-4"
            >
              <svg
                class="w-8 h-8 text-gray-400"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="384"
                  rx="48"
                  stroke-linejoin="round"
                  stroke-width="32"
                  width="416"
                  x="48"
                  y="80"
                />
                <circle
                  cx="296"
                  cy="232"
                  r="24"
                />
                <circle
                  cx="376"
                  cy="232"
                  r="24"
                />
                <circle
                  cx="296"
                  cy="312"
                  r="24"
                />
                <circle
                  cx="376"
                  cy="312"
                  r="24"
                />
                <circle
                  cx="136"
                  cy="312"
                  r="24"
                />
                <circle
                  cx="216"
                  cy="312"
                  r="24"
                />
                <circle
                  cx="136"
                  cy="392"
                  r="24"
                />
                <circle
                  cx="216"
                  cy="392"
                  r="24"
                />
                <circle
                  cx="296"
                  cy="392"
                  r="24"
                />
                <path
                  d="M128 48v32m256-32v32"
                  fill="none"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="32"
                />
                <path
                  d="M464 160H48"
                  fill="none"
                  stroke-linejoin="round"
                  stroke-width="32"
                />
              </svg>
            </div>
            <h3
              class="text-lg font-semibold text-gray-900 mb-2"
            >
              No Events Found
            </h3>
            <p
              class="text-gray-500 max-w-sm"
            >
              There are no upcoming events at the moment. Check back soon for exciting new events! âœ¨
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
